### DISCUSSION TITLE
GFM ignores comments

### DISCUSSION THREAD

**Discussion by ivanjaros** [markedjs/marked#3419] [Q&A] [answered]
When I enable GFM I still see comments being rendered, despite them being part of the spec:

https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#hiding-content-with-comments

**Comment by UziTech** [comment]
This [demo](https://marked.js.org/demo/?text=%3C!--%20This%20content%20will%20not%20appear%20in%20the%20rendered%20Markdown%20--%3E&options=%7B%0A%20%22async%22%3A%20false%2C%0A%20%22breaks%22%3A%20false%2C%0A%20%22extensions%22%3A%20null%2C%0A%20%22gfm%22%3A%20true%2C%0A%20%22hooks%22%3A%20null%2C%0A%20%22pedantic%22%3A%20false%2C%0A%20%22silent%22%3A%20false%2C%0A%20%22tokenizer%22%3A%20null%2C%0A%20%22walkTokens%22%3A%20null%0A%7D&version=14.0.0) seems to follow the spec.

**Reply by ivanjaros** [reply]
There is some replacing taking place https://github.com/markedjs/marked/blob/60626572f1e16d256317b40f9472faa6b5a02352/src/rules.ts#L41 but am not sure what it actually does. In my case(i only use lexer), I end up with html tag and i have to check for <!-- preffix. There is no "comment" type or no transformation into empty "space" or any other tag or complete removal of it. Technically it is html tag but I would expect GFM to parse it and remove it altogether because GFM can render html tags but ignore html comment and that is not what i am observing. I guess it's one of those lexer vs renderer behavioural differences yet again.

**Reply by UziTech** [reply]
The Lexer still creates a comment token like it should but when rendered it doesn't display anything because it is rendered like an html comment

**Reply by ivanjaros** [reply]
![obrázok](https://github.com/user-attachments/assets/eea270e4-ef26-4314-938e-31c6b6118a62)

**Reply by UziTech** [reply, marked as answer]
If you check the source of this comment you will see the html comment is still there the browser just doesn't render it.

**Reply by UziTech** [reply]
Marked follows this GFM spec https://github.github.com/gfm/#example-152

**Reply by ivanjaros** [reply]
because it is treated like html. it should not be rendered at all. i render html as escaped text to discourage users from using it and this caused the text to be rendered for me which simply surprised me because i expected marked to simply ignore it altogether. i have mitigated it by checking for said prefix so it is fine.
